softwareComponents/configuration/include/configuration/joints.hpp

```diff
@@ -25,34 +29,46 @@ using JointVisitor = atoms::Visits<
  * below for concrete instances.
  *
  * Each joint has Joint::paramCount() parameters (real value), which are stored
- * in Joint::positions. These params are, e.g., angle for rotation joint.
+ * in Joint::position. These params are, e.g., angle for rotation joint.
  */
 struct Joint: public atoms::VisitableBase< Joint, JointVisitor > {
+    using Positions = tcb::span< float >;
```

Comment:
You use `Positions` as `const Positions &` which is not how to use a `span`. Therefore I would remove the typedef entirely.